### DISCUSSION TITLE
Blade include for parsing/prepping variables used in parent Blade component

### DISCUSSION THREAD

**Discussion by asdf072** [livewire/livewire#9064] [1. Help]
I'd like to use a common include file to parse and clean element attributes in many Blade components. Like:

`included-file.blade.php`
```
@php
    $category = !empty($cat) ? $cat : null;
    $build = isset($build) & empty($build) ? null : 1;
    $class = $build ? "build build-plus" : '';
@endphp
```

`input.blade.php`
```
@include('included-file')
<div>
    <input data-category="{{$category}}" data-build="{{$build}}" class="{{$class}}" />
</div>
```

`livewire-component.php`
```
<p>
    <x-input cat="flooring" build="6" />
</p>
```
I know using `@include` takes the variables out of scope. Using native php `include()` doesn't work because of view caching (I believe). View Composers don't seem to have access to passed properties. Is there another way?

**Comment by joshhanley** [comment]
@asdf072 this seems more like a Laravel Blade question, not a Livewire question. I don't think it's going to work how you hoped, but I'm not really use how Blade `@include` functions under the hood (I've not used it for years). That being said you will still need `@props` inside each of your Blade Components otherwise it won't recognise the parameters you are trying to pass in (like `cat`).
